gapis/replay/gpu_profile.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package replay
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/capture"
+	"github.com/google/gapid/gapis/service"
+	"github.com/google/gapid/gapis/service/path"
+
+	perfetto_pb "perfetto/config"
+)
+
+// Eyeball some generous trace config parameters
+const (
+	bufferSizeKb                            = uint32(256 * 1024)
+	durationMs                              = 30000
+	gpuRenderStagesDataSourceDescriptorName = "gpu.renderstages"
+)
+
+// GpuProfile replays the trace and writes a Perfetto trace of the replay
+func GpuProfile(ctx context.Context, capturePath *path.Capture, device *path.Device) (*service.GpuProfileResponse, error) {
```

Comment:
This should return `(*service.ProfilingData, error)`. You are currently wrapping things, then unwrapping them in `server.go` just to wrap them again in `grpc.go` :) I know this is because of the change I've asked for before.